be/src/exprs/vectorized/string_functions.cpp

```diff
@@ -2517,16 +2519,35 @@ int StringFunctions::index_of(const char* source, int source_count, const char*
 }
 
 struct StringFunctionsState {
+    std::string pattern;
     std::unique_ptr<re2::RE2> regex;
     std::unique_ptr<re2::RE2::Options> options;
     bool const_pattern{false};
+    phmap::flat_hash_map<int32_t, std::unique_ptr<re2::RE2>>
+            driver_regex_map; // regex for each pipeline_driver, to make it driver-local
 
     StringFunctionsState() : regex(), options() {}
+
+    // Implement a driver-local regex, to avoid lock contention on the RE2::cache_mutex
+    re2::RE2* get_or_prepare_regex() {
+        DCHECK(const_pattern);
+        int32_t driver_id = CurrentThread::current().get_driver_id();
+        if (driver_id == 0) {
+            return regex.get();
+        }
+        auto iter = driver_regex_map.find(driver_id);
```

Comment:
Is it safe call find without a mutex ?

Reply:
Mistake. I should use the thread-safe version.